src/res/mod.rs

```diff
@@ -217,11 +225,16 @@ impl Resources {
         Default::default()
     }
 
-    /// Adds a new resource to this container.
+    /// Adds a new resource to this container, or returns `r` again if the resource exists already.
```

Comment:
TBH, I'd rather see only `entry` exposed here than having 3 extra helpers for all sort of scenarios. The helper methods could be on the `Entry` itself (e.g. `or_insert_default`)